collector/lib/ConnScraper.cpp

```diff
@@ -433,7 +487,14 @@ bool ReadContainerConnections(const char* proc_path, std::vector<Connection>* co
       auto emplace_res = conns_by_ns.emplace(netns_inode, NSNetworkData());
       if (emplace_res.second) {
         auto& ns_network_data = emplace_res.first->second;
-        if (!GetConnections(dirfd, pid, &ns_network_data.connections, listen_endpoints ? &ns_network_data.listen_endpoints : nullptr)) {
+
+        ProcessInfo info;
+        ReadProcessInfo(curr->d_name, dirfd, info);
+
+        std::shared_ptr<Process> process =
+            ProcessStore::global_instance.FetchReference(Process(container_id, info.comm, info.exe, info.exe_path, info.args, pid));
```

Comment:
It seems strange that we are reading from proc and then looking for the process information from the process store. Wouldn't it make more sense to look in the process store and then read proc if the process isn't in the process store.

Reply:
This is going to change with the moving of the scraping functions to a dedicated factory. The process information will be fetched by providing only the PID.